packages/admin-ui-hooks/src/test-utils.tsx

```diff
@@ -0,0 +1,72 @@
+import 'jest-location-mock'
+
+export function jestWindowMock() {
+  let history: string[] = [window.location.pathname]
+  let index = 0
+  let popstateCallback = () => {}
+
+  Object.defineProperties(window.history, {
```

Comment:
I do not know if I understand. You say instead of putting the function directly?

Reply:
It is more about being able to undo the changes made here, and restore the original implementation of the window.history

Reply:
The package `jest-location-mock` doesn't mock the window history, just the window.location. The useQueryState on the other hand has some behaviors based on history, as it was necessary to update the param query object in scenarios where history is browsed. That's why I needed to mock the behavior to test the hook. I don't understand why the need to make it available to change these behaviors. Changing them will break the behavior of useQueryState.

Reply:
Isn't about changing these behaviors, it is about returning to original implementation.

```tsx
const unmock = jestWindowMock()

unmock() // return to original implementation
```

It is useful to clean the state, so like your `history` here, it keeps persistent between tests.
It could implicitly make something work unexpectedly between tests.
It is good for every test (`it` statements) to start from the same initial state.